src/circuit_extraction.jl

```diff
@@ -264,12 +257,16 @@ function gaussian_elimination(M::Matrix{T}, steps::Vector{GEStep} = Vector{GESte
             continue
         end
         while current_col <= nc
-            r0 = findfirst(!iszero, M[i:nr, current_col])
-            if r0 !== nothing
+            rs = findall(!iszero, M[i:nr, current_col])
+            if length(rs) > 0
+                sort!(rs, by = k -> sum(M[k,:]), rev = rev)
+                r0 = rs[1]
                 r0 += i - 1
                 r0 == i && break
-                M[i,:] = M[i,:] .⊻ M[r0,:]
-                step = GEStep(:addto, r0, i)
+                M_r0 = M[r0,:]
```

Comment:
maybe should use `@views`?

Reply:
This will swap two rows. So we have to record one of the rows. Hence, `@view` is not the right way.